serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-grpc-services.adoc

```diff
@@ -1 +1,163 @@
-= Orchestration of gRPC based services
\ No newline at end of file
+= Orchestration of gRPC based services
+
+Learn how to invoke a gRPC from a Serverless Workflow running a quarkus. This guide covers:
+
+* Creating a workflow that invokes a running gRPC service. 
+* Setting up properties to locate the running service. 
+
+== Prerequisites
+
+It's expected you have read how to create-your-first-workflow-service.html[Create Your First Workflow Guide]. 
+
+It's assumed you are familiar with main https://grpc.io/docs/what-is-grpc/core-concepts/[gRPC] concepts. 
+
+== Solution
+
+In this guide, we'll go through a Serverless Workflow application that greets a user in their preferred language by invoking a grpc service.
+
+The application is located in the `serverless-workflow-greeting-client-rpc-quarkus`
+https://github.com/kiegroup/kogito-examples/tree/main/kogito-quarkus-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus[directory].
+
+== Architecture
+
+There are following main components of this workflow application:
+
+* https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-server-rpc-quarkus/src/main/proto/greeting.proto[grpc proto file]
+* https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/src/main/resources/jsongreet.sw.json[worfklow definition file]
+* https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/pom.xml[Maven POM]
+* https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/src/main/resources/application.properties[Application properties]
+
+=== gRPC proto file
+
+This file defines a greeter service consisting of a sayHello method. This method accepts two parameters: the name of the user and the optional language to use in the greet message (English by default). The method returns the greeting message in the proper language.
+
+----
```

Comment:
@akumar074 do you know if we can add language snippets? Maybe we are missing *.proto files? 🤔

Reply:
yeah, I guess we could have a snippet for it.